internal/htmlForm/htmlForm.go

```diff
@@ -0,0 +1,106 @@
+package htmlForm
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"golang.org/x/net/html"
+)
+
+// HtmlForm represents needed elements of an HTML Form.
+type HtmlForm struct {
+	// Action parsed from action form's field
+	Action string
+	// URL is the complete url where form will be posted
+	URL string
+	// Values contains htmlForm values to be submitted
+	Values url.Values
+}
+
+// ParseForms parses and returns all htmlForm elements beneath node.  Form values
+// include all input, select with the first possible value and submit buttons. The values of radio
+// and checkbox inputs are included only if they are checked.
+func ParseForms(node *html.Node) (forms []*HtmlForm) {
+	if node == nil {
+		return nil
+	}
+
+	doc := goquery.NewDocumentFromNode(node)
+	doc.Find("form").Each(func(_ int, s *goquery.Selection) {
+		form := &HtmlForm{Values: url.Values{}}
+		form.Action, _ = s.Attr("action")
+
+		s.Find("input").Each(func(_ int, s *goquery.Selection) {
+			name, _ := s.Attr("name")
+			if name == "" {
+				return
+			}
+
+			typ, _ := s.Attr("type")
+			typ = strings.ToLower(typ)
+			_, checked := s.Attr("checked")
+			if (typ == "radio" || typ == "checkbox") && !checked {
+				return
+			}
+
+			value, _ := s.Attr("value")
+			form.Values.Add(name, value)
+		})
+		// if form has no inputs it's not interesting to us
+		_, ok := s.Find("input").Attr("type")
```

Comment:
Can't get it. Why do we need this check? And why do we have str. 35-50 before but not after this check?